crypto/x509/internal.h

```diff
@@ -575,6 +575,11 @@ OPENSSL_EXPORT int validate_cidr_mask(CBS *cidr_mask);
 
 OPENSSL_EXPORT int cn2dnsid(ASN1_STRING *cn, unsigned char **dnsid, size_t *idlen);
 
+// Match reference identifiers starting with "." to any sub-domain.
+// This is a non-public flag, turned on implicitly when the subject
+// reference identity is a DNS name.
+#define _X509_CHECK_FLAG_DOT_SUBDOMAINS 0x8000
```

Comment:
Although this one needs to be in an internal header file, it would be nice if all of the relevant bit-flags were documented here with it:
```
// External supported flags:
// * X509_CHECK_FLAG_ALWAYS_CHECK_SUBJECT 0x0001
// * X509_CHECK_FLAG_NO_WILDCARDS         0x0002
// * X509_CHECK_FLAG_NEVER_CHECK_SUBJECT  0x0020
// * X509_CHECK_FLAG_NO_PARTIAL_WILDCARDS 0x000?

// Internal flags:
// * _X509_CHECK_FLAG_DOT_SUBDOMAINS      0x8000

// External unsupported flags:
// * X509_CHECK_FLAG_MULTI_LABEL_WILDCARDS
// * X509_CHECK_FLAG_SINGLE_LABEL_SUBDOMAINS
```